jib-core/src/main/java/com/google/cloud/tools/jib/async/AsyncDependencies.java

```diff
@@ -80,4 +82,19 @@ public AsyncDependencies addSteps(List<? extends AsyncStep<?>> asyncSteps) {
   public <C> ListenableFuture<C> whenAllSucceed(Callable<C> combiner) {
     return Futures.whenAllSucceed(futures).call(combiner, listeningExecutorService);
   }
+
+  /**
+   * Calls {@code combiner} when all the added futures succeed.
+   *
+   * @param combiner the {@link Callable}
+   * @param <C> the return type of {@code combiner}
+   * @return the result returned from calling {@code combiner}
+   * @throws ExecutionException if {@code combiner} threw an exception
+   * @throws InterruptedException if the current thread was interrupted while waiting for {@code
+   *     combiner}
+   */
+  public <C> C whenAllSucceedBlocking(Callable<C> combiner)
+      throws InterruptedException, ExecutionException {
+    return Futures.whenAllSucceed(futures).call(combiner, listeningExecutorService).get();
```

Comment:
I think this looks fine, but this looks like it's just `whenAllSucceed` with a call to `.get`. I'm wondering if we can leave off the extra method and just have the `.get` explicitly in the individual functions so we get rid of the nested `ListenableFuture`s but don't need this extra method?